core/lib/zksync_core/src/api_server/web3/tests/mod.rs

```diff
@@ -472,20 +471,16 @@ impl HttpTest for LogFilterChangesWithBlockBoundaries {
         let new_events: Vec<_> = new_events.iter().collect();
 
         let bounded_logs = client.get_filter_changes(bounded_filter_id).await?;
-        let FilterChanges::Logs(bounded_logs) = bounded_logs else {
+        let FilterChanges::Hashes(bounded_logs) = bounded_logs else {
             panic!("Unexpected getFilterChanges output: {:?}", bounded_logs);
         };
-        assert!(bounded_logs
-            .iter()
-            .all(|log| log.block_number.unwrap() < 3.into()));
+        assert!(bounded_logs.is_empty());
 
         let upper_bound_logs = client.get_filter_changes(upper_bound_filter_id).await?;
-        let FilterChanges::Logs(upper_bound_logs) = upper_bound_logs else {
+        let FilterChanges::Hashes(upper_bound_logs) = upper_bound_logs else {
             panic!("Unexpected getFilterChanges output: {:?}", upper_bound_logs);
         };
-        assert!(upper_bound_logs
-            .iter()
-            .all(|log| log.block_number.unwrap() < 3.into()));
+        assert!(upper_bound_logs.is_empty());
 
         let lower_bound_logs = client.get_filter_changes(lower_bound_filter_id).await?;
```

Comment:
Please simplify logic a bit here as well. Previously, `start_idx` would need to be defined to filter out old logs, but now it can be set to 0; i.e., you can call

```rust
assert_logs_match(&lower_bound_logs, &new_events);
```